### PR TITLE
Added CropException class

### LINKED ISSUE #332: [Feat] - Adding different type of Exception for crop cancel.
Is it possible to add some CropCancellationException instead Exception(msg) here?
```
 object CancelledResult : CropImageView.CropResult(
        originalBitmap = null,
        originalUri = null,
        bitmap = null,
        uriContent = null,
        error = Exception("cropping has been cancelled by the user"),
        cropPoints = floatArrayOf(),
        cropRect = null,
        wholeImageRect = null,
        rotation = 0,
        sampleSize = 0
    )
```
 then it would be more convenient to differentiate it from other exceptions using is operator instead ==

### PR DESCRIPTION
**Pull request by rempawl**
(https://github.com/CanHub/Android-Image-Cropper/issues/332)
close #332

## Description:
added CropCancellationException and two BitmapExceptions.

## Check list for the Code Reviewer:
* [x] CHANGELOG
* [ ] README
* [ ] Wiki
* [ ] Version Number

### CODE REVIEW

File: cropper/src/main/java/com/canhub/cropper/CropException.kt [hunk 1]
```diff
@@ -0,0 +1,10 @@
+package com.canhub.cropper
+
+import android.net.Uri
+
+sealed class CropException(message: String) : Exception(message) {
+    class CancellationException : CropException("cropping has been cancelled by the user")
+    class FailedToLoadBitmapException(uri: Uri, message: String?) :
+        CropException("Failed to load sampled bitmap: $uri\r\n$message")
+    class FailedToDecodeImageException(uri: Uri) : CropException("Failed to decode image: $uri")
```
Comment (Canato) [inline cropper/src/main/java/com/canhub/cropper/CropException.kt:9]: Love it! let's just add a prefix to all messages, like `crop: cropping has been cancelled by the user`

Small amount of libraries use this, but helpful to debug!

File: CHANGELOG.md [hunk 2]
```diff
@@ -10,6 +10,8 @@ The format is based on [Keep a Changelog](https://keepachangelog.com/en/1.0.0/)
 - `Security` in case of vulnerabilities.
 
 ## [x.x.x] - unreleased
+### Added
```
Comment (Canato) [inline CHANGELOG.md:13]: Let's tag as `Changed` we change the type of return exception making it more clear

**Review by Canato — APPROVED** [review]
Great piece of work, thanks for it!

File: cropper/src/main/java/com/canhub/cropper/CropException.kt (hunk 1 above)
Comment (rempawl) [reply cropper/src/main/java/com/canhub/cropper/CropException.kt]: ok it's done

File: CHANGELOG.md (hunk 2 above)
Comment (rempawl) [reply CHANGELOG.md]: done

**Review by Canato — APPROVED** [review]
Great work, thanks!

**Pull request merged**